integration/binpack/binpack.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package binpack
+
+import (
+	"fmt"
+	"sort"
+)
+
+type timing struct {
+	name string
+	time float64
+}
+
+// we'll need to regenerate this list time to time
+var timings = []timing{
```

Comment:
I got them via running `make integration` and using your test timing printouts, manually setting the top 20 to top 500

Reply:
>  the duration of a test depends on which tests ran before

How come?

Reply:
I don't expect it to be a perfect split, but it's a good start, we can play with the numbers further or the partitioning logic...

But now the kind tests are faster than kokoro: 
![image](https://user-images.githubusercontent.com/635073/82100633-067d0d00-96bf-11ea-8fa0-49bf5459eb15.png)

And they are not that unbalanced...
![image](https://user-images.githubusercontent.com/635073/82100650-13016580-96bf-11ea-9925-c508fe0b78b3.png)

I'll review the numbers based on the outputs on travis - my numbers were from my glinux machine

Reply:
Each test that builds something benefits from the caching of artefacts built in previous tests. See how the tests used to run in 20 minutes and now they run in 30 minutes. At least, they finish earlier, *if* there are enough workers in the build pool. When we have multiple PRs fighting for workers, I think we'll have to wait more to get the results.

Reply:
In theory, we could build better partitions by grouping tests that build the same thing. Very tedious to maintain though.

Reply:
To some extent that grouping exists - because I partition only on the high level - not on the individual testcase level - I think that kind of cross-caching might be more of an issue on individual testcase level. 

The overhead of 30 mins comes from the kind overhead - it takes 5 mins for all of these to just download the skaffold builder image and spin up kind!!! I think we should start to look at minikube, with precaching the image download on the travis instance, we might be able to speed up that overhead as well.

Reply:
It's a very fair point about the worker bottleneck. I'm okay to see if we need to upgrade our subscription.

Reply:
This is the baseline time when tests are not even run (I had a mistaken error for this run where all tests were skipped): 

![image](https://user-images.githubusercontent.com/635073/82102999-3f6cb000-96c6-11ea-92e6-aa0cba693a11.png)